react-components/src/architecture/base/renderTarget/CommandsController.test.ts

```diff
@@ -0,0 +1,498 @@
+      expect(onEscapeKeyMock).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe('Test update', () => {
+    test('Should all commands when the command controller updates', () => {
+      const mockCommand = new MockCommand();
+      const mockTool = createMockTool1();
+      controller.add(mockCommand);
+      controller.add(mockTool);
+
+      const updateTester1 = new UpdateTester(mockCommand);
+      const updateTester2 = new UpdateTester(mockTool);
+
+      expect(updateTester1.CalledTimes).toBe(0);
+      expect(updateTester2.CalledTimes).toBe(0);
+      controller.update();
+      expect(updateTester1.CalledTimes).toBe(1);
+      expect(updateTester2.CalledTimes).toBe(1);
+    });
+
+    test('Should not update wnen disposed', () => {
+      const mockCommand = new MockCommand();
+      controller.add(mockCommand);
+
+      const updateTester = new UpdateTester(mockCommand);
+
+      expect(updateTester.CalledTimes).toBe(0);
+      controller.dispose();
+      controller.update();
+      expect(updateTester.CalledTimes).toBe(0);
+    });
+  });
+
+  describe('Test some event with dispatch the event on the dom element', () => {
+    test('Should invoke on correct shortcut', () => {
+      const mockCommand = new MockCommand();
+      controller.add(mockCommand);
+
+      const invokeMock = vi.fn();
+      mockCommand.invoke = invokeMock;
+
+      domElement.dispatchEvent(createKeyboardEvent('A'));
+      expect(invokeMock).toHaveBeenCalledTimes(1);
+      domElement.dispatchEvent(createKeyboardEvent('A', false));
+      expect(invokeMock).toHaveBeenCalledTimes(1);
+    });
+
+    test('Should invoke on correct shortcut', () => {
+      const mockTool = createMockTool1();
+      controller.setActiveTool(mockTool);
+
+      const onFocusChangedMock = vi.fn();
+      mockTool.onFocusChanged = onFocusChangedMock;
+
+      domElement.dispatchEvent(new FocusEvent('focus'));
+      domElement.dispatchEvent(new FocusEvent('blur'));
+
+      expect(onFocusChangedMock).toHaveBeenNthCalledWith(1, true);
+      expect(onFocusChangedMock).toHaveBeenNthCalledWith(2, false);
+    });
+
+    test('Should call onWhen when WheelEvent', () => {
+      const mockTool = createMockTool1();
+      controller.setActiveTool(mockTool);
+
+      const onWheelMock = vi.fn();
+      mockTool.onWheel = onWheelMock;
+
+      domElement.dispatchEvent(new WheelEvent('wheel'));
+      expect(onWheelMock).toHaveBeenCalledTimes(1);
+    });
+
+    test('Should call onWhen when WheelEvent', () => {
+      const mockTool = createMockTool1();
+      controller.setActiveTool(mockTool);
+
+      const onWheelMock = vi.fn();
+      mockTool.onHover = onWheelMock;
+
+      domElement.dispatchEvent(createMouseHoverEvent());
+      expect(onWheelMock).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe('Test some event with the event function at the controller itself', () => {
+    test('Should call correction function on the correct tool on hover event', async () => {
```

Comment:
```suggestion
    test('Should call correct function on the correct tool on hover event', async () => {
```
The same goes for the rest of the tests in this section